packages/amplify-category-function/src/index.ts

```diff
@@ -176,10 +176,13 @@ export function isMockable(context: any, resourceName: string): IsMockableRespon
     };
   }
   const { service, dependsOn } = resourceValue;
-  const dependsOnLayers = dependsOn
-    .filter(dependency => dependency.category === 'function')
-    .map(val => _.get(context.amplify.getProjectMeta(), [val.category, val.resourceName]))
-    .filter(val => val.service === ServiceName.LambdaLayer);
+  let dependsOnLayers;
+  if (Array.isArray(dependsOn) && dependsOn.length !== 0) {
+    dependsOnLayers = dependsOn
+      .filter(dependency => dependency.category === 'function')
+      .map(val => _.get(context.amplify.getProjectMeta(), [val.category, val.resourceName]))
+      .filter(val => val.service === ServiceName.LambdaLayer);
+  }
```

Comment:
LGTM! Personally I'd use a ternary operator instead of the `if` like this:

```ts
  const dependsOnLayers = Array.isArray(dependsOn) ? dependsOn
    .filter(dependency => dependency.category === 'function')
    .map(val => _.get(context.amplify.getProjectMeta(), [val.category, val.resourceName]))
    .filter(val => val.service === ServiceName.LambdaLayer) : [];
```

That's just preference though. What do you think?